source/MapDetailPanel.cpp

```diff
@@ -322,6 +355,28 @@ bool MapDetailPanel::RClick(int x, int y)
 
 
 
+void MapDetailPanel::GeneratePlanetCards(const System &system)
+{
+	set<const Planet *> shown{};
+	
+	planetCards.clear();
+	MapPlanetCard::clear();
+	for(const StellarObject &object : system.Objects())
+		if(object.HasSprite() && object.HasValidPlanet())
+		{
+			// The same "planet" may appear multiple times in one system,
+			// providing multiple landing and departure points (e.g. ringworlds).
+			const Planet *planet = object.GetPlanet();
+			if(planet->IsWormhole() || !planet->IsAccessible(player.Flagship()) || shown.count(planet))
+				continue;
+
+			planetCards.emplace_back(*new MapPlanetCard(object, player.HasVisited(*planet)));
```

Comment:
```suggestion
			planetCards.emplace_back(object, player.HasVisited(*planet));
```

Memory leak.

Reply:
I'll change it but could you explain further? You mean that if the planet card isn't create it'll cause a problem?

Reply:
You're creating a new object with `new` and then copying it into the vector. This newed object needs to be `delete`d manually but since you never do that it creates a memory leak. Hope that makes sense.

Reply:
Oh so clearing the vector does not delete all the objects in it if they are in it by value, I thought it did. Thanks!